### PR TITLE
Implement stats-cleanup again.

### PR DESCRIPTION
**Pull request by Raynos**
This was reverted from master because it broken hyperbahn

Hyperbahn PR: https://github.com/uber/hyperbahn/pull/93

r: @jcorbin @rf

### CODE REVIEW

**Comment by Raynos** [comment]
Tests will fail without hyperbahn PR linked.

File: channel.js [hunk 1]
```diff
@@ -253,14 +218,31 @@ function TChannel(options) {
...
+        } else {
+            self.batchStats = new BatchStatsd({
+                logger: self.logger,
+                timers: self.timers,
+                statsd: self.statsd,
+                baseTags: self.options.statTags
+            });
+            self.batchStatsAllocated = true;
+
+            self.batchStats.flushStats();
```
Comment (ShanniLi) [inline channel.js]: why this is necessary?

**Comment by ShanniLi** [comment]
other than the test is failing, lgtm.

File: lib/statsd.js [hunk 2]
```diff
@@ -20,46 +20,199 @@
...
 }
 
-TChannelStatsd.prototype.onStat = function onStat(stat) {
+BatchStatsd.prototype.pushStat = function pushStat(name, type, value, tags) {
+    var self = this;
+
+    tags.app = self.baseTags.app;
+    tags.host = self.baseTags.host;
+    tags.cluster = self.baseTags.cluster;
+    tags.version = self.baseTags.version;
```
Comment (ShanniLi) [inline lib/statsd.js]: Do we really need these: cluster, version, etc.?

**Pull request merged**